services/datamanager/builtin/builtin.go

```diff
@@ -214,13 +225,15 @@ func (svc *builtIn) initializeOrUpdateCollector(
 			if updateCaptureDir {
 				targetFile, err := datacapture.CreateDataCaptureFile(svc.captureDir, captureMetadata)
 				if err != nil {
-					return nil, err
+					return nil, captureConfigsChanged, err
 				}
+				captureConfigsChanged = true
 				collector.SetTarget(targetFile)
 			}
-			return &componentMetadata, nil
+			return &componentMetadata, captureConfigsChanged, nil
 		}
 
+		captureConfigsChanged = true
 		// Otherwise, close the current collector and instantiate a new one below.
 		collector.Close()
 	}
```

Comment:
(see other comments for context) To simplify things a bit, can remove the `captureConfigsChanged` variable from being tracked/returned in this method. You can keep the `updateCaptureDir || toggledCaptureOn` -> `svc.sessionID = uuid.NewString()` in the Update method but also have a `svc.sessionIDUpdated` bool. If that is false, here you would set:
```
svc.sessionID = uuid.NewString()
svc.sessionIDUpdated = true
```

This can be reset to false at the end of Update.